SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h

```diff
@@ -60,6 +76,46 @@ inline bool CompareSchemaObjects(const Schema_Object* Lhs, const Schema_Object*
 	return FMemory::Memcmp(LhsBuffer.Get(), RhsBuffer.Get(), Length) == 0;
 }
 
+inline bool CompareSchemaComponentData(Schema_ComponentData* Lhs, Schema_ComponentData* Rhs)
+{
+	return CompareSchemaObjects(Schema_GetComponentDataFields(Lhs), Schema_GetComponentDataFields(Rhs));
+}
+
+inline bool CompareSchemaComponentUpdate(Schema_ComponentUpdate* Lhs, Schema_ComponentUpdate* Rhs)
+{
+	if (!CompareSchemaObjects(Schema_GetComponentUpdateFields(Lhs), Schema_GetComponentUpdateFields(Rhs)))
+	{
+		return false;
+	}
+
+	return CompareSchemaObjects(Schema_GetComponentUpdateEvents(Lhs), Schema_GetComponentUpdateEvents(Rhs));
+}
+
+inline bool CompareSchemaComponentRefresh(const CompleteUpdateData Lhs, const CompleteUpdateData Rhs)
+{
+	if (!CompareSchemaObjects(Schema_GetComponentDataFields(Lhs.Data), Schema_GetComponentDataFields(Rhs.Data)))
+	{
+		return false;
+	}
+
+	if (Lhs.Events == nullptr)
+	{
+		if (Lhs.Events == Rhs.Events)
+		{
+			return true;
+		}
+
+		return Schema_GetWriteBufferLength(Rhs.Events) == 0;
```

Comment:
yeah, this is kind of an awkward one. schema opjects will never have a null event list. But the CompleteUpdate object in SpatialView could have a null event list. In which case it would be the same as an empty event list. This is why we only do the check here as it is a special case for the CompleteUpdate/ComponentRefresh object

Reply:
This is because of an implementation detail in the view delta. We could try and fix it there but it complicates the view delta for the sake of making probably only this code nicer.

Reply:
Component changes currently only point to data which is in the op list. However a complete update is guaranteed to involve a component data object but there may be no update object for an events list.
E.g. when receiving ops: RemoveComponent, AddComponent.